`make buildoss` build failure on 22.1.5

**Describe the problem**

I'm trying to build an OSS build of v22.1.5.  I'll post fuller details below.  In my environment, I first saw this failure:

```
docgen http \
--gendoc ./bin/protoc-gen-doc \
--out docs/generated/http \
--protoc-flags "-Ipkg -I./vendor/github.com/gogo/protobuf -I./vendor/go.etcd.io -I./vendor/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis -I./vendor/github.com/cockroachdb/errors -I./vendor/github.com/prometheus/client_model ./pkg/server/serverpb/status.proto ./pkg/server/serverpb/admin.proto ./pkg/server/status/statuspb/status.proto"
touch bin/.docgen_http
docgen logformats docs/generated/logformats.md
touch bin/.docgen_logformats
Hash: db7417ee41f9729a2871
Version: webpack 4.46.0
Time: 81155ms
Built at: 08/17/2022 10:24:23 PM
 1 asset
Entrypoint vendor = vendor.oss.dll.js
  [47] (webpack)/buildin/global.js 472 bytes {0} [built]
 [207] (webpack)/buildin/module.js 497 bytes {0} [built]
[1068] dll vendor 12 bytes {0} [built]
[1069] ../cluster-ui/dist/js/main.js 7.69 MiB {0} [built]
[1078] ./src/js/protos.js 11.2 MiB {0} [built]
[1232] /home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/moment/locale sync ^\.\/.*$ 3.21 KiB {0} [built]
    + 1972 hidden modules

ERROR in dll vendor
Module not found: Error: Can't resolve '@cockroachlabs/crdb-protobuf-client-ccl' in '/home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach/pkg/ui/workspaces/db-console'
 @ dll vendor vendor[2]
gmake: *** [Makefile:1380: pkg/ui/workspaces/db-console/dist/vendor.oss.dll.js] Error 2
gmake: Leaving directory '/home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach'
```

It seemed a little suspicious to me that it's trying to pull in a file with `ccl` in it for an OSS build but I haven't tracked down what's going on here.  My environment is a little weird (I'm building on illumos with a few local patches) so I figured I'd try with the stock builder and I got a different problem:

```
Killed
error Command failed with exit code 137.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "build:bundle" exited with 137.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [Makefile:363: pkg/ui/cluster-ui/dist/main.js] Error 1
```

**To Reproduce**

My system:

```
$ uname -a
Darwin zathras.local 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64
$ go version
go version go1.19 darwin/amd64
```

I know this is newer than Cockroach's Go, but I'm using the Docker builder so I didn't think that mattered.

My path and clone setup:

```
$ go env GOPATH
/Users/dap/go
$ pwd
/Users/dap/go/src/github.com/cockroachdb/cockroach
$ git status
HEAD detached at v22.1.5
nothing to commit, working tree clean
```

Start the builder and run the build.  The output is too long to put in this description so I've [attached it](https://github.com/cockroachdb/cockroach/files/9376180/build-failure.txt).

**Expected behavior**

I expected the "buildoss" build to complete successfully and produces binaries.


**Environment:**

- Cockroach v22.1.5 (from Git)
- MacOS 11.6 (Big Sur)
- Docker Desktop 2.2.0.3

Maybe related: #83528, #69101, #75137?

Jira issue: CRDB-18742

Indeed, `make` is still supported for now.

I'm a little perplexed about this bit:

```
Module not found: Error: Can't resolve '@cockroachlabs/crdb-protobuf-client-ccl' in '/home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach/pkg/ui/workspaces/db-console'
 @ dll vendor vendor[2]
gmake: *** [Makefile:1380: pkg/ui/workspaces/db-console/dist/vendor.oss.dll.js] Error 2
gmake: Leaving directory '/home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach'
```

So it seems like the `make` build thinks your workspace is at `/home/dap/garbage-compactor/cockroach/cache/gopath/src/github.com/cockroachdb/cockroach`, but according to your report your `GOPATH` is `/Users/dap/go` and `pwd` is `/Users/dap/go/src/github.com/cockroachdb/cockroach`. Can you clarify?

Good call!  I think it was set to 2 GiB before.  I set it to 8 GiB and got a new failure:

```
build/node-run.sh -C pkg/ui/workspaces/cluster-ui yarn build
yarn run v1.22.19
$ npm-run-all -p build:typescript build:bundle
$ tsc
$ NODE_ENV=production webpack --display-error-details --mode=production
/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack-cli/bin/cli.js:93
				throw err;
				^

Error: Cannot find module './DependenciesBlock'
Require stack:
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/Module.js
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/NormalModule.js
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/NormalModuleFactory.js
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/Compiler.js
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/webpack.js
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack-cli/bin/utils/validate-options.js
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack-cli/bin/utils/convert-argv.js
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack-cli/bin/cli.js
- /go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/bin/webpack.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/Module.js:9:27)
    at Module._compile (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/NormalModule.js:19:16)
    at Module._compile (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/NormalModuleFactory.js:17:22)
    at Module._compile (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/Compiler.js:23:29)
    at Module._compile (/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/Module.js',
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/NormalModule.js',
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/NormalModuleFactory.js',
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/Compiler.js',
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/lib/webpack.js',
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack-cli/bin/utils/validate-options.js',
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack-cli/bin/utils/convert-argv.js',
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack-cli/bin/cli.js',
    '/go/src/github.com/cockroachdb/cockroach/pkg/ui/node_modules/webpack/bin/webpack.js'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "build:bundle" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [Makefile:363: pkg/ui/cluster-ui/dist/main.js] Error 1
make: *** Waiting for unfinished jobs....
```

[Here is the full log this time.](https://github.com/cockroachdb/cockroach/files/9377883/build-issue-docker-8gib.txt)

Heyyyy @davepacheco  — sorry about that!  I'm not entirely sure how that slipped by us, but I'm able to confirm the original error you reported (`Module not found: Error: Can't resolve '@cockroachlabs/crdb-protobuf-client-ccl' in (some path)`) on my normal Linux dev machine.  Looks like it snuck past me during the 22.1 release while I was getting the Bazel build working!

https://github.com/cockroachdb/cockroach/pull/86425 should take care of things for you.  If you're feeling particularly generous and wouldn't mind confirming that that PR fixes your build, I'd super appreciate it 🙂

--- a/pkg/ui/workspaces/db-console/webpack.vendor.js
+++ b/pkg/ui/workspaces/db-console/webpack.vendor.js
@@ -15,7 +15,7 @@ const webpack = require("webpack");
 
 const pkg = require("./package.json");
 
-const prodDependencies = Object.keys(pkg.dependencies);
+const prodDependencies = Object.keys(pkg.dependencies).filter(name => !name.startsWith("@cockroachlabs"));
 
 // tslint:disable:object-literal-sort-keys
 module.exports = (env, argv) => {

